### ISSUE TITLE
100

### ISSUE THREAD

**Issue by regielya** [olivierkes/manuskript#819]

**Comment by TheJackiMonster** [comment]
Maybe it is using spell-checking which usually applies on opening texts. One major problem is definitely that most functionality in manuskript is still single-threaded. So implementing multi-threading for tasks which parse huge text files could enhance the application overall.

**Comment by marosoft** [comment]
I am not sure how the spell checking is implemented.
I think it would make sense to not to do it just after opening a text but few seconds later.
Another thing that could be done is to do the spell checking of the current paragraph only and have an option to do it for the whole text adhoc.

**Comment by TheJackiMonster** [comment]
@marosoft Yes, I addressed this on another issue (#280). Basically we could put the spell checking into a separate thread so it wouldn't slow down the GUI an its interaction.

**Comment by TheWithz** [comment]
If spell checking is a slow operation, then would it make sense to cache the checks? Presumably, if the text is already written and spellcheck was on, then the checks were already performed, and because the text hasn't changed, then there's no reason to recalculate the information.

**Comment by TheJackiMonster** [comment]
@TheWithz That's exactly what I implemented in the new spell checker using languagecheck. Unfortunately there are still some other issues regarding that spellchecker (see #832) but I'm going to fix these hopefully soon. I can also look into the other spell checking (most common is enchant probably) to add a cache if not implemented yet.

Maybe we can even adapt the spell checking to iterate over parts of the text only while editing. So the iteration over the full text has to be done only once when loading the file.

**Comment by tustin2121** [comment]
Did anything come of this? Because I'm having this issue too. Turning off spell check makes changing text fast, but I kind of need spellcheck...
